jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java

```diff
@@ -476,6 +477,7 @@ else if (networkChannel instanceof DatagramChannel)
                 selectorManager.accept(channel, context);
             else
                 selectorManager.connect(channel, context);
+                listener.onConnectBegin((SocketChannel)channel);
```

Comment:
The "begin" event should be emitted just before attempting to connect.
Therefore, this line should be moved in the 2 places that attempt to connect (blocking and non-blocking), and pass the `address` as second parameter.